src/extra/gfx/main_window.cpp

```diff
@@ -85,6 +85,9 @@ struct MainWindow : public Base {
   }
 
   SHTypeInfo compose(SHInstanceData &data) {
+    // Require extra stack space for the wire containing the main window
+    data.wire->stackSize = std::max<size_t>(data.wire->stackSize, 4 * 1024 * 1024);

```

Comment:
Not sure this is good. At least should be a constant in `foundation.hpp` or so.

Reply:
Does it need to be that big, or could it also work with a smaller value (like `512 * 1024`)?

Reply:
It doesn't that matter that much how big (this 4MB) since it'll only get applied to the wire containing MainWindow (of which there is only 1 in normal cases)

Reply:
I'd rather avoid cryptic bugs than setting it too low and reducing application memory footprint by like 2 megabytes.

Reply:
Yeah ok, and yeah 4MB is ok 👌 
Just consider linker by default has 1MB set on windows. But it's true that some apps (likely game engines etc might go higher.)